include/stdx/rollover.hpp

```diff
@@ -0,0 +1,155 @@
+    }
+    [[nodiscard]] constexpr friend auto operator!=(rollover_t lhs,
+                                                   rollover_t rhs) -> bool {
+        return not(lhs == rhs);
+    }
+
+    [[nodiscard]] constexpr friend auto operator<(rollover_t lhs,
+                                                  rollover_t rhs) -> bool {
+        constexpr auto mid = std::numeric_limits<underlying_t>::max() / 2;
+        return static_cast<underlying_t>(lhs.value - rhs.value) > mid;
+    }
+    [[nodiscard]] constexpr friend auto operator>(rollover_t lhs,
+                                                  rollover_t rhs) -> bool {
+        return rhs < lhs;
+    }
+    [[nodiscard]] constexpr friend auto operator<=(rollover_t lhs,
+                                                   rollover_t rhs) -> bool {
+        return not(rhs < lhs);
+    }
+    [[nodiscard]] constexpr friend auto operator>=(rollover_t lhs,
+                                                   rollover_t rhs) -> bool {
+        return not(lhs < rhs);
+    }
+
+#if __cpp_lib_three_way_comparison >= 201711L
+    [[nodiscard]] constexpr friend auto operator<=>(rollover_t lhs,
+                                                    rollover_t rhs)
+        -> std::compare_three_way_result_t<underlying_t> {
+        using cmp_t = std::compare_three_way_result_t<underlying_t>;
+        if (lhs.value == rhs.value) {
+            return cmp_t::equal;
+        }
+        return lhs < rhs ? cmp_t::less : cmp_t::greater;
+    }
+#endif
+
+    [[nodiscard]] constexpr friend auto
+    operator+(rollover_t lhs, rollover_t rhs) -> rollover_t {
+        lhs += rhs;
+        return lhs;
+    }
+    [[nodiscard]] constexpr friend auto
+    operator-(rollover_t lhs, rollover_t rhs) -> rollover_t {
+        lhs -= rhs;
+        return lhs;
+    }
+    [[nodiscard]] constexpr friend auto
+    operator*(rollover_t lhs, rollover_t rhs) -> rollover_t {
+        lhs *= rhs;
+        return lhs;
+    }
+    [[nodiscard]] constexpr friend auto
+    operator/(rollover_t lhs, rollover_t rhs) -> rollover_t {
+        lhs /= rhs;
+        return lhs;
+    }
+    [[nodiscard]] constexpr friend auto
+    operator%(rollover_t lhs, rollover_t rhs) -> rollover_t {
+        lhs %= rhs;
+        return lhs;
+    }
+
+    underlying_t value{};
+};
+
+template <typename T> rollover_t(T) -> rollover_t<std::make_unsigned_t<T>>;
+
+} // namespace v1
+} // namespace stdx
+
+template <typename T>
+struct std::numeric_limits<stdx::rollover_t<T>> : std::numeric_limits<T> {
+    constexpr static auto min() noexcept -> stdx::rollover_t<T> {
+        return stdx::rollover_t<T>{std::numeric_limits<T>::min()};
+    }
+    constexpr static auto lowest() noexcept -> stdx::rollover_t<T> {
+        return stdx::rollover_t<T>{std::numeric_limits<T>::lowest()};
+    }
+    constexpr static auto max() noexcept -> stdx::rollover_t<T> {
+        return stdx::rollover_t<T>{std::numeric_limits<T>::max()};
+    }
+};
```

Comment:
question: do these make sense? are they required?

Reply:
I think you're right, these don't make sense. I might have originally provided them to work with `std::chrono`, but I'm not sure. I'll investigate.